### PR TITLE
Auto update parquets

### PR DESCRIPTION
**Pull request by nup002**
This pull request adds new functionality:

- Optional direct update of parquet files
Parquet files can be updated directly without the need for intermediate CSV files. This makes it much more convenient for users who have downloaded the parquet files from kaggle and wish to keep the up to date. 
- Improved output
A progressbar is displayed during the fetching of candles, showing the user how many candles will be retrieved, and how long it is expected to take.
image

- Additional user settings
The user can set global variables on the top of Main.py to control how the script behaves. 
image

The code has also been refactored in order to more easily implement the changes listed above. The functionality of the code has not been changed, and the script can be used just as before this pull request.

### CODE REVIEW

**Comment by nup002** [comment]
I have noticed a few minor issues that I will attempt to correct. Please wait with the merge.

**Comment by gosuto-inzasheru** [comment]
Wooooow this has been on my list for so long (#13), really appreciate this! I wouldn't even mind getting rid of the csv files completely—they just take up space. I will wait for your final PR and dig through the code. Impression so far is awesome!

**Comment by nup002** [comment]
All right, it's good to go!

**Comment by nup002** [comment]
Happy you like it. A few more issues to iron out.

**Comment by nup002** [comment]
Okay, it's ready on my end. There were no additional issues that had to be fixed.

File: main.py [hunk 1]
```diff
@@ -76,121 +89,246 @@ def get_batch(symbol, interval='1m', start_time=0, limit=1000):
...
+            print(f"Will fetch {missing_data_timedelta} ({total_minutes_of_data} minutes) worth of candles.")
+            if in_pycharm: time.sleep(0.2)
+            first_read = False
+            if total_minutes_of_data >= BATCH_SIZE*2:
+                bar = ProgressBar(max_value=total_minutes_of_data).start()
+
+        if bar is not None:
+            time_covered = datetime.fromtimestamp(last_timestamp / 1000) - start_datetime
+            minutes_covered = int(time_covered.total_seconds()/60)
+            bar.max_value = max(int((datetime.now() - start_datetime).total_seconds()/60), minutes_covered)
```
Comment (gosuto-inzasheru) [inline main.py:163]: @nup002 can I ask what you are doing here? It seems to introduce a bug where the max of the bar can become higher than what is communicated in line 154 (`total_minutes_of_data`).

Wouldn't it be better for consistency to stick to `max(bar.max_value, minutes_covered)`?

Comment (nup002) [reply main.py:163]: Yes, `max(bar.max_value, minutes_covered)` is a much more elegant solution.

**Pull request merged**